src/nile/utils/get_nonce.py

```diff
@@ -0,0 +1,35 @@
+"""Retrieve nonce for a contract."""
+import logging
+import os
+import subprocess
+
+from nile.common import get_gateway
+
+GATEWAYS = get_gateway()
+
+
+def get_nonce(contract_address, network):
+    """Get the current nonce for contract address in a given network."""
+    nonce = get_nonce_without_log(contract_address, network)
+
+    logging.info(f"Current Nonce: {nonce}")
```

Comment:
i wonder whether we should print this here or directly on the CLI, for consistency with other commands. the only counter argument i can think of is that i've been thinking that having the ability to turn on/off nile's logging from nre could be useful, i'd lean towards opt-in rather than opt-out logging though, but not sure.

Reply:
I should mention, Nile's `logging` doesn't print for the NRE (I assume it's lost in the abstraction, but I'd have to look deeper to give a better answer). If we want the output from these functions, I know `print` returns the output through the NRE. I'm sure there's a better way with `logging` though (if indeed we want this)

Reply:
Currently, other commands are printing from the command code, not from the CLI, so I added the logging here for consistency.

To disable logging from nre, right now is as easy as running `logging.disable(logging.INFO)` right before executing the call to the command. This is opt-out, but we could add opt-in by adding a param in each nre command default to True,  disabling the logging like this (ERROR messages would still be logged, but maybe this is a expected behavior, if not we can disable all logging using `logging.disable(logging.CRITICAL)`).

Reply:
that's true i got confused by the fact that nre commands today don't print by default, but it should be handled from logging not print. so i guess it's a matter of turning it on? that'd be the opt-in way i was thinking about, and i was thinking for it to be global for all commands, although an interface to do it per-command is not a bad idea.

the only doubt that remains to me is whether we should make this turning on/off logging independent of logging's global config, to not interfere with the user's script in case they use the module too directly or through other library.

Reply:
I think we can get the specific logger, and disable from there instead of the global logging object, similar to what we are doing in run.py for scripting. Not sure if we should name this logger (I would need to go a little bit deeper in the logging library), but I think we can achieve what we want like this.

Reply:
Also, I noticed some print statements in some commands, should we refactor those to logging too?

Reply:
Similar to @ericnordelo's solution, we could just include a `verbose` flag in `run`. I think this will follow the opt-in approach without too much refactoring i.e. `nile run scripts/deploy.py --verbose`.

Something like this:

```
def run(path, network, verbose):
    """Run nile scripts passing on the NRE object."""
    logger = logging.getLogger()
    if not verbose:
        logger.disabled = True
    script = SourceFileLoader("script", path).load_module()
    nre = NileRuntimeEnvironment(network)
    script.run(nre)
```

Further, if users use another lib or use the modules too closely, they can set up their own log handling. I think keeping it in for outside cases is the more flexible approach.

Reply:
I agree, but I think it would be helpful to name the logger so that the user can silence or activate logs specifically for the package from other integrations, I'm not sure how this works, but to silence logs from urllib3 requests, I can run `logging.getLogger("urllib3").setLevel(logging.WARNING)`, we can do something similar.

Reply:
I created a separated issue for this.